### PR TITLE
Cronscheduler

### PR DESCRIPTION
**Pull request by ghost**
#### What type of PR is this?

/kind bug
/kind regression

#### What this PR does / why we need it:

This PR fixes the issue where nextScheduleTimeDuration sends -ve time if multiple previous executions are skipped because of issues. Because of the issue cron job controller is scheduling the cron job constantly which is incorrect and a side effect of it is KCM publishing too many logs and wasted CPU cycles.

The logic change for the function was done in https://github.com/kubernetes/kubernetes/pull/97098 but the issue became prevalent once this function is used for calculating the next schedule duration in https://github.com/kubernetes/kubernetes/pull/109250. This PR reverts few changes made [here](https://github.com/kubernetes/kubernetes/commit/1240a29af9cf0d33fe8d3b5f655ff91038f0d75b) to get count of missed schedules and if there are any missed schedules we are assigning mostRecentTime to now to avoid sending -ve mostRecentTime.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
NONE

```release-note
Fixes a regression in 1.25+ handling of failed/stuck jobs/pods in the cronjob controller.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sushanth0910** [comment]
@kow3ns @krmayankk could you please review this PR.

**Comment by liggitt** [comment]
please update the release note with the exact release(s) the regression was introduced

**Comment by sushanth0910** [comment]
>please update the release note with the exact release(s) the regression was introduced
@liggitt i have updated it

File: pkg/controller/cronjob/utils.go [hunk 1]
```diff
@@ -160,13 +142,18 @@ func mostRecentScheduleTime(cj *batchv1.CronJob, now time.Time, schedule cron.Sc
...
 	} else if mostRecentTime == nil {
-		// no missed schedules since earliestTime
-		mostRecentTime = &earliestTime
+		if numberOfMissedSchedules > 0 {
+			// if there are missed schedules since earliestTime
+			mostRecentTime = &now
+		} else {
+			// no missed schedules since earliestTime
+			mostRecentTime = &earliestTime
+		}
```
Comment (soltysh) [inline pkg/controller/cronjob/utils.go:156]: This above condition is the critical piece we want back, but I'm still not inclined to leak that misleading information about number of missed schedules out of `mostRecentScheduleTime`. So I'm proposing alternative approach in https://github.com/kubernetes/kubernetes/pull/121327

**Review by soltysh — CHANGES_REQUESTED** [review]
/hold 
let's sync about the approach

File: pkg/controller/cronjob/utils.go (hunk 1 above)
Comment (sushanth0910) [reply pkg/controller/cronjob/utils.go:156]: /lgtm

**Comment by sushanth0910** [comment]
issue fixed by an alternate approach https://github.com/kubernetes/kubernetes/pull/121327

**Pull request closed without merging**